src/dataframe/dataframe.jl

```diff
@@ -234,7 +236,9 @@ function Base.getindex(df::DataFrame, col_ind::ColumnIndex)
 end
 
 # df[MultiColumnIndex] => (Sub)?DataFrame
-function Base.getindex{T <: ColumnIndex}(df::DataFrame, col_inds::AbstractVector{T})
+function Base.getindex{T <: ColumnIndex}(df::DataFrame,
+                                         col_inds::Union{AbstractVector{T},
+                                                         AbstractVector{Nullable{T}}})
```

Comment:
I guess I added these because it triggered failures in tests. The problem is that any `DataArray` was accepted by this signature, but that's no longer the case with `NullableArrays`. This means you can't index a data frame based on a condition on one of its columns. I think that's a showstopper.